### PR TITLE
Validate feature dependencies on feature startup

### PR DESCRIPTION
**Pull request by mikedennis**
Add a hook for a dependency check when running node feature executor. Throw exception if a feature's dependency is missing.

Fix for issue #285

### CODE REVIEW

File: Stratis.Bitcoin/Builder/Feature/FeaturesExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        /// <summary>
+        /// Ensures a dependency feature type is present in the feature registration.
+        /// </summary>
+        /// <typeparam name="T">The dependency feature type.</typeparam>
+        /// <param name="features">List of feature registrations.</param>
+        /// <returns>List of feature registrations.</returns>
+        /// <exception cref="MissingDependencyException">Thrown if feature type is missing.</exception>
+        public static IEnumerable<IFullNodeFeature> EnsureFeature<T>(this IEnumerable<IFullNodeFeature> features)
+        {
+            if (!features.Any(i => i.GetType() == typeof(T)))
```
Comment (dangershony) [inline Stratis.Bitcoin/Builder/Feature/FeaturesExtensions.cs]: Can you not do?
```
!features.OfType<T>().Any()
```

File: Stratis.Bitcoin/Features/Miner/MiningFeature.cs [hunk 2]
```diff
@@ -1,56 +1,66 @@
...
+        ///<inheritdoc />
+        public override void Stop()
+        {
+        }
 
-		public override void Stop()
-		{
-		}
-	}
+        ///<inheritdoc />
```
Comment (dangershony) [inline Stratis.Bitcoin/Features/Miner/MiningFeature.cs]: Excellent this looks very nice.

File: Stratis.StratisD/Program.cs [hunk 3]
```diff
@@ -32,6 +33,7 @@ public static void Main(string[] args)
                 .UseStratisConsensus()
                 .UseBlockStore()
                 .UseMempool()
+                .UseWallet()
```
Comment (dangershony) [inline Stratis.StratisD/Program.cs]: Hmm, I am not sure we need wallet in the daemon.

Comment (mikedennis) [inline Stratis.StratisD/Program.cs]: The daemon currently is built with .AddPowPosMining() 
which has the dependency on the wallet manager. Should we remove PowPosMining then? Or have another extension method that adds the wallet dependencies that mining needs?

File: Stratis.Bitcoin/Builder/Feature/FeaturesExtensions.cs (hunk 1 above)
Comment (mikedennis) [inline Stratis.Bitcoin/Builder/Feature/FeaturesExtensions.cs]: Sure - will give that a try!

File: Stratis.StratisD/Program.cs (hunk 3 above)
Comment (dangershony) [inline Stratis.StratisD/Program.cs]: ah right yes so keep it

**Comment by mikedennis** [comment]
Ok, I've fixed up the type checking code. I like that much better, thanks for the suggestion!

I see that there is a Stratis.Bitcoin.Features.LightWallet feature now that Jeremy has been working on.  I wonder whether it has enough of the wallet capability for the Miner feature? I see it does include the WalletManager. Potentially we could use that instead of the full blown Wallet? Let me know if you think this is worthwhile investigating and I can create another issue for it and investigate it.

**Pull request merged**